src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.java

```diff
@@ -37,8 +37,12 @@
     extends AbstractViolationReporter
     implements FileSetCheck {
 
-    /** Collects the error messages. */
-    private final SortedSet<LocalizedMessage> messageCollector = new TreeSet<>();
+    /**
+     * Collects the error messages.
+     * @noinspection ThreadLocalNotStaticFinal
+     */
+    private final ThreadLocal<SortedSet<LocalizedMessage>> messageCollector =
```

Comment:
It doesn't sound like it is expected behavior since the original field wasn't `static` either which would have produced the same effect.

But I don't see any negative side effects changing this to have all messages in a single place.
@romani What do you think about this?